### DISCUSSION TITLE
View all forms in the public GraphQL schema not working

### DISCUSSION THREAD

**Discussion by paulnzime** [solspace/craft-freeform#1450] [Questions & Answers] [answered]
### What happened?

If I create a Public GraphQL schema in Craft, I am given the options to select each of my forms to be viewable but also an option for "View all forms"

If I tick the view all forms but leave the others unchecked, then I'd expect to be able to query on the public API for any of my forms, instead nothing is returned.

The main issue here being that in production we disallow admin changes, which then hides the schema options. This means if our clients create forms in production, we cannot enable them to be available in GraphQL.

### How can we reproduce this?

1.  Edit or enable the Public GraphQL schema
2. Tick the box for View all forms leaving any forms created unchecked
3. Using GraphiQL, the following query using this new schema will return nothing:
```
query ($handle: [String]) {
  freeform {
    form(handle: $handle) {
      name
    }
  }
}
```
4. The same query will work fine when using Full Schema.

### Freeform Edition

Pro

### Freeform Version

5.5.3

### Craft Version

5.2.5

### When did this issue start?

Fresh install of Freeform

**Comment by seandelaney** [comment, marked as answer]
@paulnzime 

I'm struggling to replicate what you are seeing.

I've setup multiple forms and ticked "View all forms" under GQL Public Schema.

![image](https://github.com/user-attachments/assets/2dc268a3-bc16-4b97-9965-74405579d561)

I can see these forms when I query forms or an individual form. I've done this Full Schema and Public Schema.

![image](https://github.com/user-attachments/assets/ec0e681b-30e4-420e-889e-810331af2547)

I've also disabled allow admin changes and added another new form and refreshed GQL query explorer and ran my query again and the newly create form is listed in the results.

If I look at the DB / Project Config, I can see View all forms is listed.

![image](https://github.com/user-attachments/assets/67ff95f6-f6ec-4e02-9ddb-3bd2eecd9c1d)

Can you give me more info on how your Schema is setup so I can replicate and try reproduce the issue?

Thanks

**Comment by seandelaney** [comment]
@paulnzime Is there anything under User Permissions causing the issue?

![image](https://github.com/user-attachments/assets/3494d18a-0f09-419a-9fbe-0b69f6ad4acf)

**Comment by paulnzime** [comment]
@seandelaney Many thanks for the detailed response, I've had another look at this and can confirm it works if **only** view all forms is checked. So in the following I can query for Enquiry but not for Test under the public schema.

![Screenshot 2024-08-01 at 15 45 54](https://github.com/user-attachments/assets/fdc81874-7b1b-4cc8-b6a7-50fec5caf230)

Untick Enquiry and I can now query for them all.

I guess this kind of makes sense, its either all forms OR the ones ticked. At some point this must have become ticked on our clients website leading to this issue.

Thanks for your help.
